recipes-app/switchserialmode/files/src/serial/cp2102n.c

```diff
@@ -29,6 +29,8 @@
 #define CP2102N_RS485_HOLD_INDEX       671
 #define MAIN_RESET_LATCH_P1_INDEX      587
 #define CP2102N_GPIO2_RESET_LATH_BIT   5
+#define DEFAULT_RS485_SETUP_TIME 5
+#define DEFAULT_RS485_HOLD_TIME  5
```

Comment:
The default value is decided by the PHY, not the CP2102n itself, so better to refine to some compiling time provided env, such env could be defined in the bb file at this moment, and in the future could be decidable according to the board variant.

Reply:
```c
#ifdef BOARD_SETUP_TIME
#define DEFAULT_RS485_SETUP_TIME    BOARD_SETUP_TIME
#else
#define DEFAULT_RS485_SETUP_TIME    (0x05)
#endif
```

same for `DEFAULT_RS485_HOLD_TIME`

Reply:
> The default value is decided by the PHY, not the CP2102n itself, so better to refine to some compiling time provided env, such env could be defined in the bb file at this moment, and in the future could be decidable according to the board variant.

correct, this value is related to what's your transceiver.
But I don't think this is a good way to pass the default value from 
bb file.
Imagine that one board type has two different transceivers because of material shortage.

if in that case, I think we need more consideration of the software structures.

Reply:
Then add back the platform abstraction.

Reply:
It can't.  We need more infomaration to bind the controller and transceiver currently. But we lack this.
We'd like to provide a recommended value for most of the scenes.  It can be adjustable. So 0x5 is just our recommendation
If this is not reasonable, so give the user a recommendation that needs to be removed.

Reply:
> It can't. We need more infomaration to bind the controller and transceiver currently. But we lack this.

Such as?

Reply:
> > Imagine that one board type has two different transceivers because of material shortage.
> 
> From software point of view, if these two xcers are not compatible, then they are not the same `one board type`.
just a second resource,  there is no need to add one board type.
I think this modification is OK for the current

Reply:
I am not talking about the board type stored on the device, but rather a `logic` board type that only meaningful for the switching serial mode scenario.